src/main/scala/no/ndla/articleapi/caching/Memoize1.scala

```diff
@@ -9,36 +9,57 @@
 
 package no.ndla.articleapi.caching
 
+import java.util.concurrent.{ScheduledThreadPoolExecutor, TimeUnit}
+
 import no.ndla.articleapi.ArticleApiProperties.ApiClientsCacheAgeInMs
 
-class Memoize1[T, R](maxCacheAgeMs: Long, f: T => R) extends (T => R) {
+class Memoize1[T, R](maxCacheAgeMs: Long, f: T => R, autoRefreshCache: Boolean) extends (T => R) {
   case class CacheValue(value: R, lastUpdated: Long) {
     def isExpired: Boolean = lastUpdated + maxCacheAgeMs <= System.currentTimeMillis()
   }
 
   private[this] val cache = scala.collection.mutable.Map.empty[T, CacheValue]
 
+  private def renewCache(v1: T) = {
+    cache.put(v1, CacheValue(f(v1), System.currentTimeMillis()))
+  }
+
+  if (autoRefreshCache) {
+    val ex = new ScheduledThreadPoolExecutor(1)
+    val task = new Runnable {
+      def run() = {
+        cache.keys.map(renewCache)
+      }
+    }
+    ex.scheduleAtFixedRate(task, 20, maxCacheAgeMs, TimeUnit.MILLISECONDS)
+  }
+
   def apply(v1: T): R = {
     cache.get(v1) match {
+      case Some(cachedValue) if autoRefreshCache => cachedValue.value
       case Some(cachedValue) if !cachedValue.isExpired => cachedValue.value
       case _ => {
-        cache.put(v1, CacheValue(f(v1), System.currentTimeMillis()))
-        cache.get(v1).get.value
+        renewCache(v1)
+        cache(v1).value
       }
     }
   }
 
 }
 
 object Memoize1 {
-  def apply[T, R](f: T => R) = new Memoize1[T, R](ApiClientsCacheAgeInMs, f)
+  def apply[T, R](f: T => R) = new Memoize1[T, R](ApiClientsCacheAgeInMs, f, autoRefreshCache = false)
 }
 
-class Memoize[R](maxCacheAgeMs: Long, f: () => R) extends(() => R) {
-  private[this] val cache = new Memoize1[Option[String], R](maxCacheAgeMs, x => f())
+class Memoize[R](maxCacheAgeMs: Long, f: () => R, autoRenewCache: Boolean = false) extends(() => R) {
```

Comment:
Hvorfor to klasser? (`Memoize` og `Memoize1`). Jeg ser signaturforskjellen, men hva er behovet?